File: src/refresh.py
# 1. Read and print a list
print("--- Challenge 1: Read and Print a List ---")
my_list = ["Train", "Bus", "Tram"]
for item in my_list:
    print(f"Transport option: {item}")
print()

# 2. Count words in a sentence
print("--- Challenge 2: Count Words in a Sentence ---")
sentence = "JourneyGuard makes rail travel easy and safe"
words = sentence.split()
print(f"Sentence: '{sentence}'")
print(f"Word Count: {len(words)}\n")

# 3. Reverse a string
print("--- Challenge 3: Reverse a String ---")
text = "Automation"
reversed_text = text[::-1]
print(f"Original: {text}")
print(f"Reversed: {reversed_text}\n")

# 4. Handle a simple exception with try/except
print("--- Challenge 4: Try/Except Exception Handling ---")
try:
    print("Attempting to divide 10 by 0...")
    result = 10 / 0
except ZeroDivisionError:
    print("Caught an error: You cannot divide a number by zero!")

File: src/Refresh.java
public class Refresh {
    public static void main(String[] args) {
        // 1. Print numbers 1-100
        System.out.println("--- Challenge 1: Numbers 1 to 100 ---");
        for (int i = 1; i <= 100; i++) {
            System.out.print(i + " ");
        }
        System.out.println("\n");

        // 2. Reverse a string
        System.out.println("--- Challenge 2: Reverse a String ---");
        String original = "JourneyGuard";
        String reversed = new StringBuilder(original).reverse().toString();
        System.out.println("Original: " + original);
        System.out.println("Reversed: " + reversed + "\n");

        // 3. Check Even or Odd
        System.out.println("--- Challenge 3: Even or Odd ---");
        int num = 42;
        if (num % 2 == 0) {
            System.out.println(num + " is Even\n");
        } else {
            System.out.println(num + " is Odd\n");
        }

        // 4. Largest number in an array
        System.out.println("--- Challenge 4: Largest Number in Array ---");
        int[] arr = {12, 45, 78, 23, 56};
        int max = arr[0];
        for (int n : arr) {
            if (n > max) {
                max = n;
            }
        }
        System.out.println("The largest number is: " + max);
    }
}